### PR TITLE
feat(kubernetes): added deployment name to pod resource id

### PR DESCRIPTION
**Pull request by PelegLi**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

## Description

concatenate the Deployment resource's name to the nested Pod's resource id and adding it to the Pod's config in `_parent_resource`

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my feature, policy, or fix is effective and works
- [x] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by bo156 — APPROVED** [review]

**Review by ChanochShayner — APPROVED** [review]

File: checkov/kubernetes/graph_builder/local_graph.py [hunk 1]
```diff
@@ -147,6 +147,9 @@ def _extract_nested_resources_recursive(conf: Dict[str, Any], all_resources: Lis
         if not template or not isinstance(template, dict):
             all_resources.append(conf)
             return
+        if conf.get('kind', {}) == 'Deployment':
```
[the remark this answers is not in the document]
Comment (PelegLi) [reply checkov/kubernetes/graph_builder/local_graph.py:150]: no specific reason, can be removed

File: checkov/kubernetes/graph_builder/local_graph.py [hunk 2]
```diff
@@ -147,6 +147,9 @@ def _extract_nested_resources_recursive(conf: Dict[str, Any], all_resources: Lis
         if not template or not isinstance(template, dict):
             all_resources.append(conf)
             return
+        if conf.get('kind') == 'Deployment':
+            # means this is a Pod resource nested in a Deployment resource
+            template[PARENT_RESOURCE_KEY_NAME] = conf.get('metadata', {}).get('name', "")
```
Comment (gruebel) [inline checkov/kubernetes/graph_builder/local_graph.py:152]: the idea was to add the resource ID of the parent, then it is easier to query it later.
```python
template[PARENT_RESOURCE_ID_KEY_NAME] = ...
```

**Review by gruebel — COMMENTED** [review]
nice work, but it would be better to add the actual resource ID for the parent reference 🙂

File: checkov/kubernetes/graph_builder/local_graph.py [hunk 3]
```diff
@@ -147,6 +147,10 @@ def _extract_nested_resources_recursive(conf: Dict[str, Any], all_resources: Lis
         if not template or not isinstance(template, dict):
             all_resources.append(conf)
             return
+        if conf.get('kind') == 'Deployment':
+            # means this is a Pod resource nested in a Deployment resource
+            template[PARENT_RESOURCE_ID_KEY_NAME] = get_resource_id(conf)
```
Comment (gruebel) [inline checkov/kubernetes/graph_builder/local_graph.py:152]: 🥇

**Pull request merged**